ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -19629,6 +19628,7 @@ bool HqlCppTranslator::needsRealThor(IHqlExpression *expr, unsigned flags)
     case no_globalscope:
     case no_extractresult:
     case no_pure:
+    case no_datasetfromrow:
```

Comment:
Why did this move? Just to confuse future reviewers?

Reply:
This is the change that corresponds to "Commit message suggests dataset(row) has changed but code change does not look like it would".  It has moved from a set of case labels that return false to a set of case labels that recurse into checking the input parameter,

Reply:
Maybe it's a github display bug, but it looks in the comparison as if it's moved further down in same list of cases